### PR TITLE
[Android] Added ability to specify `largeIcon` as URL.

### PR DESCRIPTION
**Pull request by tschoffelen**
Not only can you now specify `largeIcon` as a bundle item, but when the value starts with `http`, it gets downloaded, rounded, and set as the large icon of the notification.

### CODE REVIEW

**Comment by zorosun** [comment]
@tschoffelen 

Does it work to use `file://...` path as URL ?

**Comment by tschoffelen** [comment]
Not with this version. Either `http://` or `https://`, or as a bundle string that resolves to the image path. 

Allowing `file://`would require changing [line 150](https://github.com/zo0r/react-native-push-notification/pull/468/commits/db18a3bade867420248477850ffd5b11be76d7d2#diff-daa707ee59f959fbde5b91695d70012eR150) to check for something like `://` instead.

**Comment by zorosun** [comment]
It works! thank you.

**Comment by Angelk90** [comment]
@tschoffelen: How do I place a Large notification icon, an image taken from an http url.
Example: https://batch-doc.s3.amazonaws.com/Push/BigIcon.png
Is not it very clear to me? I need that the notifications come from the server.

**Comment by tschoffelen** [comment]
That is exactly what this PR makes possible. Simply add a `largeIcon` key to your notification payload (where you also specify the notification text) with a string value for the string of the URL of the image.

**Comment by Angelk90** [comment]
@tschoffelen: So I have to change the original file with this: https://raw.githubusercontent.com/tschoffelen/react-native-push-notification/db18a3bade867420248477850ffd5b11be76d7d2/android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationHelper.java

**Comment by tschoffelen** [comment]
Yep! That should do it.

**Comment by Angelk90** [comment]
Does not work:
Server parse:
```
Parse.Push.send({
    channels: channel,
    data: {
      title: title,
      alert: text,
      objectId: postId,
      badge: "Increment",
      sound: "default",
      color:"#006ACC",
      largeIcon: "https://images.vanityfair.it/gallery/58873/Mid/248b4ee8-d0e8-4e0e-8dcc-882ca863f536.jpg"
    }
  },{
    useMasterKey: true,
    success: function(){
      console.log("We pushed!!");
    },
    error: function(error){
      console.log("Some error occurred:"+error);
    }
  });
```

I receive as notification:
immagine

Where am I wrong?

**Comment by tschoffelen** [comment]
Your server-side code seems fine. Did you remember to recompile after changing the file?

**Comment by Angelk90** [comment]
You do, you can see me: console.log ('NOTIFICATION:', notification);
What prints in detail.

**Comment by Angelk90** [comment]
And a screen of how it should be the notification I receive.

**Comment by tschoffelen** [comment]
That does not mean the Java code has been recompiled, the javascript just outputs whatever you put in the payload. The result should be exactly the way your screenshot showed, we use it in our company for that exact purpose.

**Comment by Angelk90** [comment]
@tschoffelen: How do i recompiled the java code?

**Comment by Angelk90** [comment]
I took the file it replaced and then from cmd i invoked: react-native run-android

**Comment by zorosun** [comment]
@Angelk90  are you using localNotification to push?

**Comment by Angelk90** [comment]
@zorosun: No, they are notifications that come from a server, so:
           OnNotification: function (notification) {
               Console.log ('NOTIFICATION:', notification);

**Comment by zorosun** [comment]
this is what I did

`
onNotification: async function(notification) {
      console.log('NOTIFICATION:', notification);
      if (notification.userInteraction) {
          // when use click notification from home screen
      } else {
          // onReceive from server when push notification
          // I pushed to localNotification to get what I want

          PushNotification.localNotification({
              bigText: 'message',
              subText: 'message',
              largeIcon: 'http://image_url_from_server',
              title: 'title',
              message: 'message'
           });
      }
    },
`

**Comment by Angelk90** [comment]
@zorosun: But when you are out of the application do not receive the double notification?

**Comment by zorosun** [comment]
@Angelk90 I didnt do the server but when OnNotification called my phone doesnt show anything.

**Comment by Angelk90** [comment]
@zorosun: Instead, my device shows two notifications when I'm out of the app.
@tschoffelen: How do I compile a module?

**Comment by zorosun** [comment]
react-native run-android should recompile

**Comment by Angelk90** [comment]
@zorosun: I tried with: react-native run-android should recompile
Nothing does not work either, you always see the image of the ic_launcher.

**Comment by zorosun** [comment]
not sure if you replace the file correctly but you can try 

- replace master branch with @tschoffelen's one in package.json 
- remove node_modules/react-native-push-notification
- npm install again
- react-native run-android

if it doesnt work then I have no idea

**Comment by Angelk90** [comment]
@zorosun: I took the file: https://raw.githubusercontent.com/tschoffelen/react-native-push-notification/db18a3bade867420248477850ffd5b11be76d7d2/android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationHelper.java
Replaced it with the file: node_modules \ react-native-push-notification \ android \ src \ main \ java \ com \ dieam \ reactnativepushnotification \ modules \ RNPushNotificationHelper.java
What I had, and I compiled as you told me.
but it does not work.

**Comment by Angelk90** [comment]
@zorosun: Do you think I'm wrong?
@tschoffelen: Where am I wrong?

**Comment by tschoffelen** [comment]
I think there might be some cache files (possibly in `android/build` or `android/app/build`) that still have the old version - you might try removing (or moving just to be safe) those directories and re-run the app. Also note that I think you have to fully stop the running app after the new version is installed, and then start it again to register the new notification handling code. 

If that doesn't work, look at if the RNPushNotificationHelper.java is also somewhere else (globally installed or copied by React Native), you might have patched the wrong version of the file. Telling NPM to get the package from my repo would be the easiest way to resolve this as @zorosun suggested.

I just checked - this is the exact version of the code we use in production, so the code itself definitely works.

**Comment by tschoffelen** [comment]
Sorry to the reviewers for this tangent by the way, didn't mean to mess up the PR with troubleshooting...

**Comment by Angelk90** [comment]
@tschoffelen: I've deleted app from the device.
Canceled module, and added to your:
https://github.com/tschoffelen/react-native-push-notification
Then I did: cd android && gradlew clean
But it does not work.

You could tell me what to print: Console.log ('NOTIFICATION:', notification);
Is OnNotification Empty?

File: android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationHelper.java [hunk 1]
```diff
@@ -229,30 +295,6 @@ public void sendToNotificationCentre(Bundle bundle) {
...
-                            soundName = soundName.substring(0, soundName.lastIndexOf('.'));
-                            resId = context.getResources().getIdentifier(soundName, "raw", context.getPackageName());
-                        }
-
-                        soundUri = Uri.parse("android.resource://" + context.getPackageName() + "/" + resId);
-                    }
-                }
-                notification.setSound(soundUri);
-            }
-
```
Comment (godness84) [inline android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationHelper.java]: @tschoffelen why you deleted these lines?

Comment (godness84) [reply android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationHelper.java]: I meant lines 232-255

Comment (tschoffelen) [reply android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationHelper.java]: No idea, think this must be added later or something. Added it back in.

**Comment by godness84** [comment]
Have you seen my PR that is based on yours? It adds a "largeIconCircular"
key, but mostly it fixes an issue with the rounded image when it's
rectangular and not perfectly squared.

**Comment by vvavepacket** [comment]
when is this feature going into master so that we can use https images on largeIcon ?

**Comment by vvavepacket** [comment]
@godness84 

which source code of yours contains 'largeIconCircular', because I pulled latest fork of tschoffelen  and can't find largeIconCircular anywhere. I tried putting largeIconCircular as false, but it is still rounded icon

**Comment by wesleymc** [comment]
@tschoffelen Have you already tried receiving a remote notification with the application terminated? It seems to me that the approach of your PR crash the application when it is closed and gets a notification with a largeIcon URL to deal with. Error.

**Comment by sauhardar** [comment]
Anyone know how to position the large icon (as passed by URL) to the right of the title/bodyText or how to make it round? Is there any way of styling icons received through URL?

**Pull request closed without merging**